Changes

```diff
@@ -133,13 +133,20 @@ Working version
   SUPPORTS_SHARED_LIBRARIES from Makefile.config.
   (David Allsopp, review by Gabriel Scherer and Mark Shinwell)
 
-- GPR#1733: change the perspective of the unexpected existential error message.
-  (Florian Angeletti, review by Gabriel Scherer and Jeremy Yallop)
-
 - GPR#1720: Improve error reporting for missing 'rec' in let-bindings.
   (Arthur Charguéraud and Armaël Guéneau, with help and advice
    from Gabriel Scherer, Frédéric Bour, Xavier Clerc and Leo White)
 
+- GPR#1733,1993,1998,2058,2094: Various error message improvements
+    - change the perspective of the unexpected existential error message.
+    - expanded error messages for universal quantification failure
+    - more context for unbound type parameter error
+    - full explanation for unsafe cycles in recursive module definitions
+      (suggestion by Ivan Gotovchits)
+    - rewording for "constructor has no type" error
```

Comment:
The grouping is nice (I hope people don't get the impression that those improvements were minor, recursive cycles in particular were a carefully crafted change).

I think you could point out that those are "typing error message improvements" -- as opposed to runtime errors or parsing/lexing errors.

I would still list the PR associated to each individual change (in addition to having them at the top), so that one can more easily get context/discussion on a specific change.

Reply:
Good points, I will apply those changes once the error message is fixed.